module_fcst_grid_comp.F90

```diff
@@ -25,8 +25,8 @@ module module_fcst_grid_comp
                                 operator(+), operator (<), operator (>),   &
                                 operator (/=), operator (/), operator (==),&
                                 operator (*), THIRTY_DAY_MONTHS, JULIAN,   &
-                                NOLEAP, NO_CALENDAR, date_to_string,       &
-                                get_date
+                                GREGORIAN, NOLEAP, NO_CALENDAR,            &
```

Comment:
I would suggest not change the default calendar type from Julian to Gregorian until it is fully tested in FV3 or it is confirmed by GFDL that the FV3 is working with Gregorian calendar and we can test it. So far from the previous GFDL atmosphere model driver, coupler_main.F90 we have:

        select case( uppercase(trim(calendar)) )
        case( 'JULIAN' )
            calendar_type = JULIAN
        case( 'NOLEAP' )
            calendar_type = NOLEAP
        case( 'THIRTY_DAY' )
            calendar_type = THIRTY_DAY_MONTHS
        case( 'NO_CALENDAR' )
            calendar_type = NO_CALENDAR
        case default
            call mpp_error ( FATAL, 'COUPLER_MAIN: coupler_nml entry calendar must '// &
                                    'be one of JULIAN|NOLEAP|THIRTY_DAY|NO_CALENDAR.' )
        end select

It looks to me the Gregorian calendar is not supported.

Reply:
It is supported by FMS. GREGORIAN parameter exists in FMS, it's just not added to this select case block. All data (except for calendar attributes of course) are identical in the new baseline outputs compared to previous baselines.

Reply:
How do we deal with GFSv15 restart files which had "Julian" specified in the coupler.res file for warming start and restart run, and it is not consistent with default "Gregorian" calendar. The fv3_cap reads calendar_type (line 461 in fv3_cap.F90) and the date, I am not sure if we can replace the calendar_type to "Gregorian" directly or we want to add some code changes to convert he Julian date to Gregrian date, or at lease add some warning message if there is any inconsistency? Please see lines 460-464 in fv3_cap.F90.

Reply:
Calendar type in coupler.res will be Gregorian not Julian after this change, which should be consistent. See, for example, RESTART/coupler.res in /scratch1/NCEPDEV/nems/emc.nemspara/RT/NEMSfv3gfs/ufs-public-release-20191227/fv3_gfs_v15p2_ccpp.